Follow up introduction doc (#487)
oam-dev/kubevela: docs/introduction.md

### Platform agnostic developer tools

The typical example is [Waypoint](https://github.com/hashicorp/waypoint). Waypoint is a developer facing tool which introduces a consistent workflow (i.e., build, deploy, release) to ship applications on top of different platforms.

KubeVela can be integrated into Waypoint like any other supported platforms. In this case, developers will use the Waypoint workflow instead of the KubeVela Appfile/CLI to manage applications, and all the capabilities of KubeVela including abstractions will still be available in this integration.

### Helm, etc. 

Helm is a package manager for Kubernetes that provides package, install, and upgrade a set of YAML files for Kubernetes as a unit. 

KubeVela is not a package manager. Developers are expected to use KubeVela's Appfile to describe how to build and deploy application and that Appfile is not Kubernetes YAML format. However, the server side API object behind Appfile (i.e. `Application Configuration`) is indeed a Kubernetes CRD, it could be packaged and distributed by Helm at ease.

### Kubernetes

> KubeVela is a Kubernetes plugin. But you could also think of KubeVela as a special Kubernetes "distribution" that focuses on developers and application management only. It leverages the native Kubernetes extensibility and capabilities to resolve a hard problem - making application management enjoyable on Kubernetes.

Reviewer: KubeVela is a Kubernetes plugin -> KubeVela is a Kubernetes plugin for abstraction.   Let us remove the 2nd sentence.

## Motivation

> The modern trend of cloud-native technology is moving towards pursuing consistent application delivery across clouds and on-premises infrastructures using Kubernetes as the common abstraction layer. Kubernetes, although excellent in abstracting away low-level infrastructure details, does introduce another layer of complexity to application developers, namely: pods, port exposing, privilege escalation, resource claims, CRD, and so on. We’ve seen such a learning curve and the lack of developer-facing abstraction have impacted user experience, slowed down productivity, led to unexpected errors or misconfigurations in production. People start questioning the value of this revolution: "why am I bothered with that many YAML files?" they asked.

Reviewer: - abstracting away low-level infrastructure details. (remove "away", the gramma should be abstract sth away from sth);

- namely -> namely understanding the concepts of 
- such a learning curve-> such a nontrivial learning curve;
- Remove they asked in the end of this paragraph.

On the other hand, abstracting Kubernetes to serve developers' needs is a highly opinionated process, and the resultant abstractions would only make sense had the decision makers been the platform builders. Unfortunately, the platform builders today face the following dilemma:

*There is no tool or framework for them to easily extend the abstractions if any*. 

> Thus, many platforms today introduce restricted abstractions, implementations and add-on mechanisms despite the extensibility of Kubernetes. This makes extending such platforms per developers' requirements or to wider scenarios almost impossible, not to mention taking full advantage of the extremely rich Kubernetes ecosystems. . 

Reviewer: - taking full advantage -> taking the full advantage
- remove the extra . in the end of the paragraph

> In the end, developers complain such a platform is too rigid and slows in response to feature requests or improvements. The platform engineers do want to help but the engineering cost is daunting: any simple API change in the platform could easily become a marathon negotiation around the opinionated abstraction design.

Reviewer: - such a platform is too rigid and slows -> those platforms are too rigid and slow 
- engineering cost -> engineering effort

## What is KubeVela?

> For developers, KubeVela is an easy-to-use tool that enables developers to describe and ship their applications to Kubernetes with zero effort. Instead of managing a handful Kubernetes YAML files, a simple docker-compose style [Appfile](./docs/developers/devex/appfile.md) is what all they need, following an application-centric workflow that can be easily integrated with any CI/CD pipeline.

Reviewer 2: ```suggestion
For developers, KubeVela itself is an easy-to-use tool that enables them to describe and ship their applications to Kubernetes with minimal efforts. Instead of managing a handful of Kubernetes YAML files, a simple docker-compose style [Appfile](./docs/developers/devex/appfile.md) is all they need. KubeVela also comes with an application-centric workflow that can be easily integrated with any CI/CD pipeline.
```

> 

Reviewer: - I think we should add something to link the above to the below。
`The above cannot be achievement without Kubevela's innovative offerings to platform builders. For platform builder, ...`

> For platform builders, KubeVela is an equally powerful framework which could help platform team create developer facing platforms with Kubernetes at ease. In detail, KubeVela relieves the pains of building such platforms by doing the following:

Reviewer 2: ```suggestion
For platform builders, KubeVela serves as a framework that empowers them to create developer facing Kubernetes based platforms with ease. KubeVela relieves the pains of building such platforms by adhering to the following design principals:
```

> For platform builders, KubeVela serve as a framework that empowered platform teams create developer facing yet highly extensible platforms at ease. In detail, KubeVela relieves the pains of building such platforms by doing the following:

Reviewer: - that empowered platform teams create -> that empowers platform teams to create
- In detail -> in details

- Application Centric. Behind the Appfile, KubeVela enforces the corresponding **application** concept as its main API and **ALL** KubeVela's capabilities serve for the applications' needs only. This is achieved by adopting the [Open Application Model](https://github.com/oam-dev/spec) as the main API for the platform.
 

> - Natively Extensible. An application in KubeVela is composed of various pluggable workload types and operational traits. Capabilities from Kubernetes ecosystem can be added to KubeVela as new workload types or trait through Kubernetes CRD registry mechanism at any time.

Reviewer: - Natively Extensible -> Extending natively.
- Add a link to `traits` since this term appears in the first time.
- or trait -> or traits

Author: Done. The link to "traits" will be added after `docs/concepts.md` is finished, will be sth like this: https://github.com/oam-dev/rudr/blob/master/docs/concepts/traits.md

> - Highly Extensible. An application in KubeVela is composed of various workload types and operational traits which are all addons by design. New workload types and traits can be added by platform administrators to KubeVela through Kubernetes native CRD registry mechanism at any time.

Reviewer 2: ```suggestion
- Highly Extensible. An application in KubeVela is composed of various workload types and operational traits which are all addons by design. Platform administrators can add new workload types and traits to the KubeVela platform through Kubernetes' native CRD registry mechanism at any time.
```

> - Simple yet Powerful Abstraction Layer. KubeVela's Appfile and CLI are built with [CUELang](https://github.com/cuelang/cue) as the contract between the user-facing schemas and the underline Kubernetes resources. KubeVela provides a set of built-in CUE templates to start with and platform teams are free to modify them at any time. Template and parameters changes take effect immediately, neither recompiliation nor redeployment of KubeVela is required. 

Reviewer 2: ```suggestion
- Simple yet Powerful Abstraction Layer. KubeVela's Appfile and CLI are built with [CUELang](https://github.com/cuelang/cue) as the contract between the user-facing schemas and the underline Kubernetes resources. KubeVela provides a set of built-in CUE templates to start with and platform teams are free to modify them at any time. Template and parameter changes take effect at runtime, neither recompilation nor redeployment of KubeVela is required. 
```

> With KubeVela, platform builders now finally have the tooling support to design and ship any new capabilities to their end-users with high confidence and turn around time measured in hours. 

Reviewer: - measured in hours ( is it good or not?      Without knowing the context, one cannot justify if it is impressive or not). So either choose even aggressive but intuitive expression say `measured in minutes` or just say `low turn around time`.

> With KubeVela, platform builders now have indefinite flexibilities in designing and shipping new capabilities to end users with confidence and efficiency. 

Reviewer 2: ```suggestion
With KubeVela, platform builders now finally have the tooling support to design and ship any new capabilities to their end-users with high confidence and turn around time measured in hours. 
```

## Comparisons

### Platform-as-a-Service (PaaS) 

The typical examples are Heroku and Cloud Foundry. They provides full application management capabilities and aim to improve developer experience and efficiency. KubeVela shares the same goal but its built-in features are much lighter and easier to maintain compared to most of the existing PaaS offerings. KubeVela core components are nothing but a set of Kubernetes controllers/plugins.

Though the biggest difference lies in the extensibility. 

Most PaaS systems enforce constraints in the type of supported applications and the supported capabilities. They are either inextensible or create their own addon systems maintained by the their own communities. In contrast, KubeVela is designed to fully leverage the Kubernetes ecosystem as its capability pool. Hence, there's no additional addon system is introduced in this project. A new capability can be installed in KubeVela at any time by simply registering the CRD and providing a CUElang template.

> This comparison also applies to many other platform agnostic developer tools such as `The Serverless Framework`, etc.

Reviewer: - Remove `This comparison also applies to many other platform agnostic developer tools such as `The Serverless Framework`, etc. ' ?

### Platform agnostic developer tools

The typical example is [Waypoint](https://github.com/hashicorp/waypoint). Waypoint is a developer facing tool which introduces a consistent workflow (i.e., build, deploy, release) to ship applications on top of different platforms.

KubeVela can be integrated into Waypoint like any other supported platforms. In this case, developers will use the Waypoint workflow instead of the KubeVela Appfile/CLI to manage applications, and all the capabilities of KubeVela including abstractions will still be available in this integration.

### Helm, etc. 

Helm is a package manager for Kubernetes that provides package, install, and upgrade a set of YAML files for Kubernetes as a unit. 

KubeVela is not a package manager. Developers are expected to use KubeVela's Appfile to describe how to build and deploy application and that Appfile is not Kubernetes YAML format. However, the server side API object behind Appfile (i.e. `Application Configuration`) is indeed a Kubernetes CRD, it could be packaged and distributed by Helm at ease.

> In implementation side, KubeVela heavily relies on Helm to package and manage its third-party plug-ins such as `Prometheus`, etc. 

Reviewer: I feel it will be better to remove `In implementation side, KubeVela heavily relies on Helm to package and manage its third-party plug-ins such as `Prometheus`, etc.`

### Kubernetes

> KubeVela is a Kubernetes addon. Though for platform builders, it's reasonable to think KubeVela as a special Kubernetes "distribution" that focuses on developers and application management only. It leverages the native Kubernetes extensibility and capabilities to resolve a hard problem - making application management enjoyable on Kubernetes.

Reviewer: How about just say 
`KubeVela is a special Kubernetes "distribution" that focuses on developers and application management only`

We probably need a clear msg here so saying addon, platform builders etcs make the claim unclear.
